src/Storages/MergeTree/GinIndexStore.cpp

```diff
@@ -446,30 +524,53 @@ void GinIndexStoreDeserializer::initFileStreams()
     String metadata_file_name = store->getName() + GinIndexStore::GIN_SEGMENT_METADATA_FILE_TYPE;
     String dict_file_name = store->getName() + GinIndexStore::GIN_DICTIONARY_FILE_TYPE;
     String postings_file_name = store->getName() + GinIndexStore::GIN_POSTINGS_FILE_TYPE;
+    String filter_file_name = store->getName() + GinIndexStore::GIN_FILTER_FILE_TYPE;
 
     metadata_file_stream = store->storage->readFile(metadata_file_name, {}, std::nullopt, std::nullopt);
     dict_file_stream = store->storage->readFile(dict_file_name, {}, std::nullopt, std::nullopt);
     postings_file_stream = store->storage->readFile(postings_file_name, {}, std::nullopt, std::nullopt);
+    filter_file_stream = store->storage->readFile(filter_file_name, {}, std::nullopt, std::nullopt);
 }
 void GinIndexStoreDeserializer::readSegments()
 {
     UInt32 num_segments = store->getNumOfSegments();
     if (num_segments == 0)
         return;
 
-    using GinIndexSegments = std::vector<GinIndexSegment>;
-    GinIndexSegments segments (num_segments);
-
     assert(metadata_file_stream != nullptr);
 
-    metadata_file_stream->readStrict(reinterpret_cast<char *>(segments.data()), num_segments * sizeof(GinIndexSegment));
-    for (UInt32 i = 0; i < num_segments; ++i)
+    /// TODO(ahmadov): clean up versions and V1 handling
```

Comment:
Let's keep our lifes simple for now (at least in the final version of this PR) and not add a new version. We are free to do that in the next 4 weeks.

Reply:
actually, I'm planning to remove V2 after this PR